cropduster/shortcuts.py

```diff
@@ -0,0 +1,55 @@
+import PIL.Image
+from django.contrib.contenttypes.models import ContentType
+
+from .fields import CropDusterField
+from .models import Image, Crop, Thumb
+from .resizing import Box
+
+
+def crop_instance(instance):
+    """
+    Auto-generate crops for a model instance.
+    """
+    fieldnames = [f.name for f in (instance._meta.fields + instance._meta.many_to_many) if isinstance(f, CropDusterField)]
+    for fieldname in fieldnames:
+        field = getattr(instance, fieldname)
+        sizes = field.sizes
+        if callable(sizes):
+            sizes = sizes()
+
+        original_w = field.width
+        original_h = field.height
+
+        obj_ct = ContentType.objects.get_for_model(instance)
+        crop_img, _ = Image.objects.get_or_create(content_type=obj_ct, object_id=instance.pk)
```

Comment:
Shouldn't this function only be used when there isn't already a crop? Otherwise, Chris's `regenerate_thumbs` would be more appropriate, since it doesn't erase the crop selections that had been made by the user.